### PR TITLE
Fix serial scheduler task iterator tests

### PR DESCRIPTION
**Pull request by ltseeley**
Updates the `task_iterator_multiple_nones` test to expect any error
instead of a specific error type when verifying that a new task request
was not sent; either `TryReceiveError` variant may be returned depending
on the timing of the test, and both are fine.

Also fixes some lint.

Signed-off-by: Logan Seeley <seeley@bitwise.io>

### CODE REVIEW

**Review by agunde406 — APPROVED** [review]

File: libtransact/src/scheduler/serial/execution.rs [hunk 1]
```diff
@@ -303,7 +297,7 @@ mod tests {
         fn task_iterator_send_successful_but_receive_failed() {
             let logger = init_logger();
 
-            let (core_tx, core_rx) = channel();
+            let (core_tx, _core_rx) = channel();
```
Comment (vaporos) [inline libtransact/src/scheduler/serial/execution.rs:300]: Move lint to a separate commit pls.

Comment (ltseeley) [reply libtransact/src/scheduler/serial/execution.rs:300]: Updated

**Review by agunde406 — APPROVED** [review]

**Review by vaporos — APPROVED** [review]
